asynchronous-authorization/vercel-ai-next-js/src/lib/rag/embedding.ts

```diff
@@ -1,20 +1,28 @@
 import { embed, embedMany } from 'ai';
-import { openai } from '@ai-sdk/openai';
-import { desc, gt, sql, cosineDistance } from 'drizzle-orm';
+import { cosineDistance, desc, gt, sql } from 'drizzle-orm';
 import { chunk } from 'llm-chunk';
 
 import { db } from '@/lib/db';
 import { embeddings } from '@/lib/db/schema/embeddings';
+import { openai } from '@ai-sdk/openai';
 
 const embeddingModel = openai.embedding('text-embedding-3-small');
 
 export const generateEmbeddings = async (value: string): Promise<Array<{ embedding: number[]; content: string }>> => {
   const chunks = chunk(value);
+
+  // Filter out empty or whitespace-only chunks
+  const validChunks = chunks.filter((chunk) => chunk && chunk.trim().length > 0);
+
+  if (validChunks.length === 0) {
+    throw new Error('No valid chunks found after filtering empty content');
+  }
```

Comment:
was getting error uploading my sample document here w/o this (which does have some whitspace). This seems ok to filter when using `embedMany`.

[Aeneid-Summary.txt](https://github.com/user-attachments/files/22913920/Aeneid-Summary.txt)